hostports/watcher.go

```diff
@@ -172,61 +193,73 @@ func (w *watcher) onChange(version string) error {
 		return err
 	}
 
-	for _, container := range containers {
-		network := networks[container.NetworkUUID]
-		bridge := ""
+	services, err := w.c.GetServices()
+	if err != nil {
+		return err
+	}
+
+	mangedNetworks, localRouters := getLocalManagedInfo(networks, host, services)
 
+	// For iptables filter table rules
+	for _, n := range mangedNetworks {
+		if c, ok := localRouters[n.UUID]; !ok || len(c.Ports) == 0 { // For bridged and routed
+			bridge, bridgeSubnet := getBridgeInfo(n, host)
+			newChainRules[n.UUID] = ChainRule{
```

Comment:
In fact, we shouldn't use this logic to accept the traffic hitting the host. We probably need a rule `-A CATTLE_FOWARD -d ${SUBNET} -j ACCEPT`